multiqc/utils/report.py

```diff
@@ -479,7 +489,7 @@ def data_sources_tofile():
             jsonstr = json.dumps(data_sources, indent=4, ensure_ascii=False)
             print(jsonstr.encode("utf-8", "ignore").decode("utf-8"), file=f)
         elif config.data_format == "yaml":
-            yaml.dump(data_sources, f, default_flow_style=False)
```

Comment:
To avoid globally setting `yaml.add_representer(defaultdict, Representer.represent_dict)` (which also is not liked by static typing)